src/App.tsx

```diff
@@ -1,87 +1,96 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import { remult } from "remult";
 import { Task } from "./shared/Task";
 import { TasksController } from "./shared/TasksController";
+import { observer } from "mobx-react-lite";
+import { action, makeAutoObservable, observable, runInAction } from 'mobx';
 
-const taskRepo = remult.repo(Task);
 
-async function fetchTasks(hideCompleted: boolean) {
-  return taskRepo.find({
-    limit: 20,
-    orderBy: { completed: "asc" },
-    where: { completed: hideCompleted ? false : undefined }
-  });
-}
 
 
+class Store {
+  taskRepo = remult.repo(Task);
+  readonly tasks = observable<Task>([]);
+  hideCompleted = false;
+  constructor() {
+    makeAutoObservable(this);
+  }
+  replaceTasks(t: Task[]) {
+    this.tasks.replace(t.map(t => makeAutoObservable({ ...t })));
+  }
+  async loadTasks() {
+    this.replaceTasks(await
+      this.taskRepo.find({
+        limit: 20,
+        orderBy: { completed: "asc" },
+        where: { completed: this.hideCompleted ? false : undefined }
+      }));
+  }
+  addTask() {
+    this.tasks.push(makeAutoObservable(new Task()))
+  }
+  async saveTask(task: Task) {
+    try {
+      const savedTask = await this.taskRepo.save(task);
+      runInAction(() => store.tasks[store.tasks.indexOf(task)] = savedTask);
+    } catch (error: any) {
+      alert(error.message);
+    }
+  }
+  async deleteTask(task: Task) {
+    await this.taskRepo.delete(task);
+    runInAction(() => store.tasks.remove(task));
+  }
+  async setAll(completed: boolean) {
+    await TasksController.setAll(completed);
+    this.loadTasks();
+  }
+}
+const store = new Store();
 
 function App() {
-  const [tasks, setTasks] = useState<Task[]>([]);
-  const [hideCompleted, setHideCompleted] = useState(false);
+  return <Todo store={store} />
 
+}
+const Todo: React.FC<{ store: Store }> = observer(({ store }) => {
```

Comment:
```suggestion
const Todo = observer(({ store }: { store: Store }) => {
```
TS's inference is better than annotations for react components. And you get to remove that `React` import.